core/src/main/java/org/apache/iceberg/TableMetadataUtil.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.iceberg.TableMetadata.MetadataLogEntry;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+public final class TableMetadataUtil {
+
+  private TableMetadataUtil() {
+    // Utility class
+  }
+
+  public static TableMetadata replacePaths(
```

Comment:
please add a `TestTableMetadataUtil` with some tests where metadata/prefixes can be null/empty/invalid/valid